.github/workflows/publish-to-pypi.yml

```diff
@@ -5,22 +5,37 @@ on:
     types: [released]
 
 jobs:
-  build:
-      name: Publish release to PyPI
-      env:
-        PYPI_USERNAME_STSCI_MAINTAINER: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
-        PYPI_PASSWORD_STSCI_MAINTAINER: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }}
-        PYPI_USERNAME_OVERRIDE: ${{ secrets.PYPI_USERNAME_OVERRIDE }}
-        PYPI_PASSWORD_OVERRIDE: ${{ secrets.PYPI_PASSWORD_OVERRIDE }}
-        PYPI_TEST: ${{ secrets.PYPI_TEST }}
-        INDEX_URL_OVERRIDE: ${{ secrets.INDEX_URL_OVERRIDE }}
-      runs-on: ubuntu-latest
-      steps:
+  validate:
+    name: Validate metadata
+    runs-on: ubuntu-latest
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/validate@master
```

Comment:
Not sure how useful this is.  It's a little late to be validating a tag that's already pushed to `master` upon which a Github Release has already been created (which triggers this workflow).  The semver ship has already sailed.  That said, I suppose it doesn't hurt, as it prevents someone from making a release of some bogus tag and triggering this workflow.

Speaking of which, is there a way to make sure this workflow _only_ runs from `master` or say, a release branch?  I.e. not from forks?  Or is that security already built-in somewhere that I've missed?

Reply:
I agree with your points, I mainly included this step to preserve the [existing behavior](https://github.com/spacetelescope/action-publish_to_pypi/blob/master/validate_version.py) of the `spacetelescope/action-publish_to_pypi` step. Also, since we can't easily add new steps later without having to come back and update many `publish-to-pypi.yml` files, I figured it would be good to have a custom step for this, in case we want to add other forms of validation later within the custom step, so the build can fail fast for things like style checks, dependency vulnerability scans, or whatever else, rather than adding such checks to the `build-wheel`/`build-sdist`/`publish` steps.

If we decide to remove this validation we can just make the `validate` step a no-op and avoid having to update all of these workflows.